test/schedule/data_test.exs

```diff
@@ -495,6 +496,89 @@ defmodule Schedule.DataTest do
     end
   end
 
+  describe "shape_with_stops_for_trip" do
+    setup do
+      trip = %Schedule.Trip{
+        id: "trip",
+        block_id: "block",
+        route_id: "route",
+        shape_id: "shape",
+        stop_times: [
+          %StopTime{stop_id: "stop1", time: 150, timepoint_id: "t1"},
+          %StopTime{stop_id: "stop2", time: 160, timepoint_id: "t2"}
+        ]
+      }
+
+      stops = [
+        %Stop{id: "stop1", name: "Stop 1", latitude: 42.01, longitude: -71.01},
+        %Stop{id: "stop2", name: "Stop 2", latitude: 42.02, longitude: -71.02}
+      ]
+
+      shape = %Shape{
+        id: "shape",
+        points: [
+          %Point{
+            shape_id: "shape",
+            lat: 42.413560,
+            lon: -70.992110,
+            sequence: 0
+          }
+        ]
+      }
+
+      {:ok, %{trip: trip, stops: stops, shape: shape}}
+    end
+
+    test "returns nil when no trip found", %{trip: trip, shape: shape} do
+      data = %Data{trips: %{}, shapes: %{trip.route_id => [shape]}}
+
+      assert nil == Data.shape_with_stops_for_trip(data, trip.id)
```

Comment:
Minor stylistic point, but you can also use `is_nil/1` here and other places to assert that the result is nil.